### DISCUSSION TITLE
What should share as package and share by accessing file directly in root

### DISCUSSION THREAD

**Discussion by syirilrakhulh** [vercel/turborepo#9546] [Help] [answered]
### Summary

So, it's my first time use Turborepo (and not having so many experiences using monorepo structure). I shock when create Turborepo from starter I get base tsconfig and eslint as package. I have question, should we share base config of something as package? Or we can just leave it on root and access directly? What makes difference? And what better way to reuse base config of something like tailwind, tsup, tsconfig, eslint, etc.

**Comment by anthonyshew** [comment, marked as answer]
You should be using a package for these configurations so that Turborepo can handle caching correctly. A file at the root of the project is not optimal. In fact, it can be highly problematic, and is one of a few reasons that has led to the [Boundaries RFC](https://github.com/vercel/turborepo/discussions/9435).

**Reply by anthonyshew** [reply]
You may find [this video series](https://www.youtube.com/playlist?list=PLB4dxsi2dXEgpLG_eR-ShADpVjNyvQTi3) I'm working on useful. (I apologize for the self-promotion, but the question and answer are quite on-the-nose.)

**Reply by syirilrakhulh** [reply]
Thanks @anthonyshew , cool videos and explanation, it's made me more understand about monorepo especially turborepo. 🙂
